### PR TITLE
Add trace to logentry types

### PR DESCRIPTION
**Pull request by salrashid123**
Add traceID parameter for logentries:

https://github.com/GoogleCloudPlatform/google-cloud-python/issues/5505

### CODE REVIEW

File: logging/google/cloud/logging/entries.py [hunk 1]
```diff
@@ -74,11 +74,16 @@ class _BaseEntry(object):
     :param http_request: (optional) info about HTTP request associated with
                          the entry.
 
+    :type trace: str
+    :param trace: (optional) traceid to apply to the entry.
+
+
```
Comment (tseaver) [inline logging/google/cloud/logging/entries.py]: Please move this below `resource, to match the order in the actual function definition.

File: logging/google/cloud/logging/entries.py [hunk 2]
```diff
@@ -183,17 +190,21 @@ class ProtobufEntry(_BaseEntry):
     :param http_request: (optional) info about HTTP request associated with
                          the entry
 
+    :type trace: str
+    :param trace: (optional) traceid to apply to the entry.
+
```
Comment (tseaver) [inline logging/google/cloud/logging/entries.py]: Likewise here.

File: logging/google/cloud/logging/logger.py [hunk 3]
```diff
@@ -209,16 +215,20 @@ def log_text(self, text, client=None, labels=None, insert_id=None,
 
         :type timestamp: :class:`datetime.datetime`
         :param timestamp: (optional) timestamp of event being logged.
```
Comment (tseaver) [inline logging/google/cloud/logging/logger.py]: Hmm, in this case `timestamp` needs to move up, although that issue was present before the PR.

File: logging/google/cloud/logging/logger.py [hunk 4]
```diff
@@ -251,16 +261,20 @@ def log_struct(self, info, client=None, labels=None, insert_id=None,
...
```
Comment (tseaver) [inline logging/google/cloud/logging/logger.py]: `timestamp` needs to move up again here.

File: logging/google/cloud/logging/logger.py [hunk 5]
```diff
@@ -293,12 +307,15 @@ def log_proto(self, message, client=None, labels=None, insert_id=None,
...
```
Comment (tseaver) [inline logging/google/cloud/logging/logger.py]: And here.

File: logging/tests/unit/test_logger.py [hunk 6]
```diff
@@ -189,7 +191,8 @@ def test_log_text_w_unicode_explicit_client_labels_severity_httpreq(self):
                                 labels=DEFAULT_LABELS)
 
         logger.log_text(TEXT, client=client2, labels=LABELS,
-                        insert_id=IID, severity=SEVERITY, http_request=REQUEST)
+                        insert_id=IID, severity=SEVERITY, http_request=REQUEST,
+                        trace=TRACE)
```
Comment (tseaver) [inline logging/tests/unit/test_logger.py]: It seems like `Logger.log_struct` and `Logger.log_proto` should also have a testcase exercising the `trace` argument (LL 243, 358 below).

File: logging/google/cloud/logging/logger.py [hunk 7]
```diff
@@ -544,6 +571,8 @@ def commit(self, client=None):
                 info['httpRequest'] = http_req
             if timestamp is not None:
                 info['timestamp'] = _datetime_to_rfc3339(timestamp)
+            if trace is not None:
+                info['trace'] = trace
```
Comment (tseaver) [inline logging/google/cloud/logging/logger.py]: We need coverage for this case.

File: logging/tests/unit/test_logger.py (hunk 6 above)
Comment (salrashid123) [reply logging/tests/unit/test_logger.py]: added `_trace` to

```
test_log_text_w_unicode_explicit_client_labels_severity_httpreq_trace()
test_log_struct_w_explicit_client_labels_severity_httpreq_trace()
test_log_proto_w_explicit_client_labels_severity_httpreq_trace()
```

File: logging/google/cloud/logging/logger.py (hunk 7 above)
Comment (salrashid123) [reply logging/google/cloud/logging/logger.py]: added
```
test_log_text_w_trace()
test_log_struct_w_trace()
test_log_proto_w_trace()
```

File: logging/google/cloud/logging/logger.py [hunk 8]
```diff
@@ -287,18 +301,21 @@ def log_proto(self, message, client=None, labels=None, insert_id=None,
         :param http_request: (optional) info about HTTP request associated with
                              the entry.
 
+        :type timestamp: :class:`datetime.datetime`
+        :param timestamp: (optional) timestamp of event being logged.
+
```
Comment (tseaver) [inline logging/google/cloud/logging/logger.py]: lint error: blank line contains whitespace

File: logging/google/cloud/logging/entries.py [hunk 9]
```diff
@@ -76,9 +76,13 @@ class _BaseEntry(object):
 
     :type resource: :class:`~google.cloud.logging.resource.Resource`
     :param resource: (Optional) Monitored resource of the entry
+
+    :type trace: str
+    :param trace: (optional) traceid to apply to the entry.
```
Comment (tseaver) [inline logging/google/cloud/logging/entries.py]: lint error:  trailing whitespace

File: logging/google/cloud/logging/entries.py [hunk 10]
```diff
@@ -185,15 +191,19 @@ class ProtobufEntry(_BaseEntry):
...
```
Comment (tseaver) [inline logging/google/cloud/logging/entries.py]: lint error: trailing whitespace

File: logging/tests/unit/test_logger.py [hunk 11]
```diff
@@ -155,7 +155,30 @@ def test_log_text_w_timestamp(self):
...
+        client = _Client(self.PROJECT)
+        api = client.logging_api = _DummyLoggingAPI()
+        logger = self._make_one(self.LOGGER_NAME, client=client)
+
+        logger.log_text(TEXT, trace=TRACE)
+
+        self.assertEqual(api._write_entries_called_with,
+                         (ENTRIES, None, None, None))
+
+    def test_log_text_w_unicode_explicit_client_labels_severity_httpreq_trace(self):
```
Comment (tseaver) [inline logging/tests/unit/test_logger.py]: lint error: line too long.

Given that you've added a separate testcase for `logger.log_text(...trace=...)`, we can probably just revert this one (the name, at a minimum).

File: logging/google/cloud/logging/logger.py (hunk 7 above)
Comment (tseaver) [reply logging/google/cloud/logging/logger.py]: We still don't have a testcase for `Batch.commit` with `trace` in an entry, which causes a coverage error for these two lines.

**Comment by tseaver** [comment]
@dpebot Please merge when green.

**Pull request merged**